cpp/dolfinx/common/MPI.h

```diff
@@ -203,9 +203,10 @@ std::vector<int> compute_graph_edges_nbx(MPI_Comm comm,
 /// @returns (0) local indices of my post office data and (1) the data
 /// (row-major). It **does not** include rows that are in `x`, i.e. rows
 /// for which the calling process is the post office
-template <typename T>
-std::pair<std::vector<std::int32_t>, std::vector<T>>
-distribute_to_postoffice(MPI_Comm comm, std::span<const T> x,
+template <typename U>
+std::pair<std::vector<std::int32_t>,
+          std::vector<typename std::remove_reference_t<typename U::value_type>>>
+distribute_to_postoffice(MPI_Comm comm, const U& x,
```

Comment:
Should the template type be documented.

Reply:
Not straight forward because the requirements are very limited - it can be almost anything.